src/optimizers/Adagrad.jl

```diff
@@ -0,0 +1,55 @@
+type AdagradOptimizer
+    net     :: SequentialNet
+    base_lr :: Any
+    cache  :: Any
+    iter    :: Int
+
+    function AdagradOptimizer(net::SequentialNet;  base_lr=(x -> (x<7)?0.01:0.001))
+        cache = []
+        for i = 1:length(net.layers)
+            layer = net.layers[i]
+            param = getParam(layer)
```

Comment:
I'm thinking that there might be a better way to check whether a layer is a learnable layer. Maybe you could experiment on using type information?

Reply:
I tried "typeof(layer) <: LearnableLayer" but it doesn't work. I am not sure if those layers, like DenseLayer, are declared as LearnableLayer.